pymeasure/instruments/jobinyvon/jobinyvon270m.py

```diff
@@ -0,0 +1,360 @@
+
+    _lambda_max = 1171.68  # Maximum value for the wavelength in nm
+    _max_steps = 37494  # maximum number of steps for the grating motor
+    _max_steps_slit = 1102.36  # do not know what this is for?
+
+    @property
+    def default_timeout(self):
+        """
+        Usual timeout needed for normal use.
+        """
+        return 300
+
+    @property
+    def lambda_0(self):
+        """
+        Get the wavelength when the motor just finished the initialization.
+        """
+        return self._lambda_max
+
+    @property
+    def steps_in_one_nanometer(self):
+        return self._steps_nm
+
+    @property
+    def nanometers_in_one_step(self):
+        return self._nm_step
+
+    @property
+    def micrometers_in_one_step(self):
+        return self._slit_microns_step
+
+    @property
+    def max_grating_steps(self):
+        return self._max_steps
+
+    def __init__(self, adapter, name="JY270M", **kwargs):
+
+        kwargs.update(dict(baud_rate=9600,
+                           timeout=self.default_timeout,
+                           parity=Parity.none,
+                           data_bits=8,
+                           stop_bits=StopBits.one,
+                           flow_control=ControlFlow.dtr_dsr,
+                           write_termination='',
+                           read_termination='',
+                           includeSCPI=False))
+
+        super().__init__(
+            adapter,
+            name,
+            **kwargs)
+
+    gsteps = Instrument.control(
+        'H0\r',
+        'F0,%d\r',
+        "Control the relative step displacement of the grating motor.",
+        get_process=get_steps_returns,
+    )
+
+    entrysteps = Instrument.control(
+        'j0,0\r',
+        'k0,0,%d\r',
+        "Control the relative step displacement of the entry slit.",
+        get_process=get_steps_returns,
+    )
+
+    exitsteps = Instrument.control(
+        'j0,2\r',
+        'k0,2,%d\r',
+        "Control the relative step displacement of the exit slit.",
+        get_process=get_steps_returns,
+    )
+
+    def read(self, **kwargs):
+        """
+        This function reads the answer of the spectrometer one byte at a time.
+        """
+        read = b''
+        for ind in range(100):
+            try:
+                r = self.read_bytes(1)
+                read += r
+                if r == b'\r':
+                    "Not all readings end with \r so we rely on timeouts to end the loop."
+                    break
+            except VisaIOError:
+                break
+        return read.decode()
+
+    def write_read(self, command: bytes, nread: int = 100, **kwargs):
+        """
+        This function writes a command to the spectrometer and reads the
+        answer of the spectrometer one byte at a time.
+        """
+        self.write_bytes(command)
+        read = b''
+        timeout = kwargs.get('timeout', self.default_timeout)
```

Comment:
Why don't you put the timeout in the signature (i.e. as a parameter) of `write_read`?